python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py

```diff
@@ -0,0 +1,328 @@
+
+        Returns:
+            None
+        """
+        if not await self.does_collection_exist_async(collection_name):
+            await self.database.create_collection(collection_name)
+
+    async def get_collections_async(
+        self,
+    ) -> List[str]:
+        """Gets all collection names in the data store.
+
+        Returns:
+            List[str] -- A group of collection names.
+        """
+        return await self.database.list_collection_names()
+
+    async def delete_collection_async(self, collection_name: str) -> None:
+        """Deletes a collection from the data store.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+
+        Returns:
+            None
+        """
+        await self.database[collection_name].drop()
+
+    async def does_collection_exist_async(self, collection_name: str) -> bool:
+        """Determines if a collection exists in the data store.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+
+        Returns:
+            bool -- True if given collection exists, False if not.
+        """
+        return collection_name in (await self.get_collections_async())
+
+    async def upsert_async(self, collection_name: str, record: MemoryRecord) -> str:
+        """Upserts a memory record into the data store. Does not guarantee that the collection exists.
+            If the record already exists, it will be updated.
+            If the record does not exist, it will be created.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+            record {MemoryRecord} -- The memory record to upsert.
+
+        Returns:
+            str -- The unique identifier for the memory record.
+        """
+        await self.create_collection_async(collection_name)
+
+        document: Mapping[str, Any] = memory_record_to_mongo_document(record)
+
+        update_result: results.UpdateResult = await self.database[
+            collection_name
+        ].update_one(document, {"$set": document}, upsert=True)
+
+        assert update_result.acknowledged
+        return record._id
+
+    async def upsert_batch_async(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        """Upserts a group of memory records into the data store. Does not guarantee that the collection exists.
+            If the record already exists, it will be updated.
+            If the record does not exist, it will be created.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+            records {MemoryRecord} -- The memory records to upsert.
+
+        Returns:
+            List[str] -- The unique identifiers for the memory records.
+        """
+        await self.create_collection_async(collection_name)
```

Comment:
semantic_text_memory already calls this upon save_info or save_reference. This call is redundant when used with SK.